modules/_labs/select/react/stories/stories_visualTesting.tsx

```diff
@@ -87,20 +87,36 @@ export const SelectStatesMenuOn = () => (
         {label: 'Alert', props: {error: Select.ErrorType.Alert}},
         {label: 'Error', props: {error: Select.ErrorType.Error}},
       ]}
-      columnProps={[{label: 'Default', props: {}}]}
+      columnProps={[
+        {label: 'Default', props: {}},
+        {label: 'Flipped', props: {isMenuFlipped: true}},
+      ]}
     >
-      {props => (
-        <div style={{height: 250}}>
-          <SelectBase
-            {...props}
-            onChange={() => {}} // eslint-disable-line no-empty-function
-            options={normalizedOptions}
-            focusedOptionIndex={1}
-            isMenuAnimated={false}
-            isMenuHidden={false}
-          />
-        </div>
-      )}
+      {props => {
+        const buttonRef = React.createRef<HTMLButtonElement>();
+        return (
+          <div
+            style={{
+              height: 500,
+              display: 'flex',
+              alignItems: 'center',
+              justifyContent: 'flex-start',
+            }}
+          >
+            <SelectBase
```

Comment:
I noticed the menu occasionally becomes slightly misaligned with its corresponding button if the button has a fractional position (issue filed at #728). I originally noticed this in the IE visual snapshot testing for the `Select States Menu On` story and assumed it was specific to IE, but it also affects Firefox (Chrome is unaffected).

I don't _think_ this should block the introduction of this PR into v4 (not sure how common it will be for the Select button to have a fractional position, though Selects in tables may be a reasonably common use case), but curious what others think.

Reply:
I've tweaked the positioning logic of the menu and it appears to be better now in IE/Firefox (though not perfect). [Old](https://5d854c26ba934e0020f5e98a-kvhfcczlnk.chromatic.com/?path=/story/testing-react-labs-select--select-states-menu-on) vs [New](https://5d854c26ba934e0020f5e98a-yorrfvmfvx.chromatic.com/?path=/story/testing-react-labs-select--select-states-menu-on).

However, I've also since noticed the issue seems to be dependent on high-DPI vs low-DPI screens (see #728).

Reply:
It seems like it might be related to how browsers choose to handle sub-pixel positioning like Flexbox.

Chrome seems to snap to whole numbers while Firefox allows fractional values.

Here's `getBoundingClientRect()` in Chrome:
<img width="461" alt="Screen Shot 2020-06-11 at 2 53 40 PM" src="https://user-images.githubusercontent.com/338257/84438341-9e551480-abf3-11ea-814a-7d27a6976263.png">

Here's `getBoundingClientRect()` in Firefox:
![Screen Shot 2020-06-11 at 2 53 51 PM](https://user-images.githubusercontent.com/338257/84438366-aa40d680-abf3-11ea-8ab2-dda10a867b5e.png)

Subpixel values are happening in Firefox on both normal and high-DPI displays.

Reply:
From a test of one not showing correctly:

CSS
```css
transform: translate(196px, 345px);
```

`getBoundingClientRect()`
```ts
{
  x: 195.39999389648438
  y: 309
}
```

Reply:
It appears low-DPI displays will snap to pixel values making this a rounding error.

High-DPI displays can do sub-pixel rendering in Firefox (but not Chrome). Overriding the `transform` value to the exact same value as the `x` makes it render properly in both High and low DPI in Firefox.  Perhaps there's a bug in Popper?